lib/ansible/runner/credential.rb

```diff
@@ -0,0 +1,56 @@
+module Ansible
+  class Runner
+    class Credential
+      attr_reader :auth, :base_dir
+
+      def self.new(authentication_id, base_dir)
```

Comment:
Oh, I just realized this overwrites `.new`.  Neat!

I was going to comment that we are doing a double initialize, but this clearly handles that.  Maybe document this behavior somewhere, as I am sure this will catch someone off guard as I personally don't see this as a common pattern in the Ruby world.

(Note:  "common" was specifically used here, but it isn't "unheard of".  I have seen this done before, just not often.  I have mostly seen it used in framework codebases where I would guess 95% of rubyists are not reading)

Reply:
So we've actually used this a handful of times in the past (I guess my we, I mean me and @Fryguy though :thinking:). But yeah, I agree that it's pretty weird to parse at first glance.

These are the places I know about off the top of my head:
https://github.com/ManageIQ/manageiq/blob/hammer/lib/embedded_ansible.rb#L7
https://github.com/ManageIQ/manageiq/blob/master/lib/pdf_generator.rb#L2
https://github.com/ManageIQ/linux_admin/blob/master/lib/linux_admin/network_interface.rb#L24